src/eHealth.locations/services/selected-location-factory.js

```diff
@@ -62,7 +63,7 @@ angular.module('eHealth.locations.services')
             // a more narrow selection for this level or a child level
             var allItem = {
               isAll: true,
-              name: 'all'
+              name: options.allItemName
```

Comment:
Maybe you can also add the level type, something like
`name: options.allItemName + ' ' + original.plural`

https://github.com/eHealthAfrica/locations/tree/develop/json